src/components/ChatEntry.js

```diff
@@ -1,22 +1,46 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+import { useState } from 'react';
 
 const ChatEntry = (props) => {
+  const sender = props.sender;
+  const time = props.timeStamp;
+  const body = props.body;
+  const [heart, changeHeart] = useState('🤍');
+  const changeLikeButton = () => {
+    if (heart === '🤍') {
+      changeHeart('❤️');
+    } else {
+      changeHeart('🤍');
+    }
```

Comment:
This works, but it is doing some extra work that isn't necessary. The 'liked' boolean in the prop data can be used to determine if the heart should be red or white. Every time changeLikeButton is called, it will update the message data at the app level when it calls props.changeLikeButton(..), which then triggers a re-render that will pass the updated props down to this component. At this level, it is possible to determine which heart to use based on the props that are passed in, so it's not necessary to also have a state variable in this component.